Source/Core/DolphinWX/MainNoGUI.cpp

```diff
@@ -366,6 +368,21 @@ int main(int argc, char* argv[])
 
   platform->Init();
 
+  // Shut down cleanly on SIGINT and SIGTERM
+  struct sigaction sa;
+  sa.sa_handler = [](int unused) {
+    const char message[] = "A signal was received. A second signal will force Dolphin to stop.\n";
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-result"
+    write(STDERR_FILENO, message, sizeof(message));
```

Comment:
Can you just if (... < 0) { /* ignore */ } or something like that instead? Pragmas are ugly.